fluent.syntax/fluent/syntax/serializer.py

```diff
@@ -179,30 +156,39 @@ def serialize_placeable(placeable):
 
 def serialize_expression(expression):
     if isinstance(expression, ast.StringLiteral):
-        return '"{}"'.format(expression.raw)
+        return '"{}"'.format(expression.value)
     if isinstance(expression, ast.NumberLiteral):
         return expression.value
-    if isinstance(expression, ast.MessageReference):
-        return expression.id.name
-    if isinstance(expression, ast.FunctionReference):
-        return expression.id.name
-    if isinstance(expression, ast.TermReference):
-        return '-{}'.format(expression.id.name)
     if isinstance(expression, ast.VariableReference):
         return '${}'.format(expression.id.name)
-    if isinstance(expression, ast.AttributeExpression):
-        return serialize_attribute_expression(expression)
-    if isinstance(expression, ast.VariantExpression):
-        return serialize_variant_expression(expression)
-    if isinstance(expression, ast.CallExpression):
-        return serialize_call_expression(expression)
+    if isinstance(expression, ast.TermReference):
+        return serialize_reference_expression(
+            expression,
+            id='-{}'.format(expression.id.name)
+        )
+    if isinstance(
+        expression, (
+            ast.MessageReference,
+            ast.FunctionReference,
+        )
+    ):
+        return serialize_reference_expression(expression)
     if isinstance(expression, ast.SelectExpression):
         return serialize_select_expression(expression)
     if isinstance(expression, ast.Placeable):
         return serialize_placeable(expression)
     raise Exception('Unknown expression type: {}'.format(type(expression)))
 
 
+def serialize_reference_expression(expression, id=None):
```

Comment:
I'm not tooooooo fond of this one, as it indicates a type weakness, I think. I've kept this in sync with the js impl, but there's no backing for this factorization in the the types in the AST.

Reply:
Thanks for bringing it up. I'll fix this in `fluent.js` and I'll follow-up in the PR with a port of the fix.

Reply:
I fixed this in `fluent.js` in https://github.com/projectfluent/fluent.js/pull/351 and I ported the fix over to this PR in https://github.com/projectfluent/python-fluent/pull/113/commits/07f762f2570c3e0ea9c5aa3e2bb453280b5cdc75.